### PR TITLE
Implement `serverStatus` and `dataSize` commands for SQLite

### LINKED ISSUE #2775: Implement `serverStatus` and `dataSize` commands for SQLite
### What should be done?

`serverStatus`, `dataSize` commands.

### Where?

- `sqlite/msg_serverstatus.go`;
- `sqlite/msg_datasize.go`;
- TODO comments mentioning this issue.

### Definition of Done

- both commands implemented for SQLite 
- integration/compatibility tests un-skipped.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #2775.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by chilagrow** [comment]
The response seems sensible
```
test> db.serverStatus().catalogStats
{
  collections: Long("3"),
  capped: 0,
  clustered: 0,
  timeseries: 0,
  views: 0,
  internalCollections: 0,
  internalViews: 0
}
```

```
test> db.runCommand({dataSize: "test.foo"})
{
  estimate: false,
  size: Long("4096"),
  numObjects: Long("1"),
  millis: 14,
  ok: 1
}
```

File: internal/handlers/sqlite/msg_datasize.go [hunk 1]
```diff
@@ -16,13 +16,96 @@ package sqlite
...
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, document.Command())
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	started := time.Now()
+
+	stats, err := c.Stats(ctx, new(backends.CollectionStatsParams))
+	if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseDoesNotExist) ||
```
Comment (rumyantseva) [inline internal/handlers/sqlite/msg_datasize.go:89]: Is it a possible case? I don't see `Stats` can return `ErrorCodeDatabaseDoesNotExist` or `ErrorCodeCollectionDoesNotExist`.

Comment (rumyantseva) [reply internal/handlers/sqlite/msg_datasize.go:89]: or is it a different stats?.. I think I'm lost :)

Comment (chilagrow) [reply internal/handlers/sqlite/msg_datasize.go:89]: It's a `Collection.Stats` which returns error. The newly added interface is for `Backend.Stats`.

This makes me think, perhaps we can rename them for clarity like below?
`Collection.Stats` -> `Collection.CollectionStats`
`Database.Stats` -> `Database.DatabaseStats` 
`Backend.Stats` -> `Backend.ServerStats`

File: internal/backends/backend.go [hunk 2]
```diff
@@ -134,6 +136,24 @@ func (bc *backendContract) DropDatabase(ctx context.Context, params *DropDatabas
...
+// ServerStatsParams represents the parameters of Backend.Stats method.
+type ServerStatsParams struct{}
+
+// ServerStatsResult represents the results of Backend.Stats method.
+type ServerStatsResult struct {
+	CountCollections int64
+}
+
+// Stats returns statistics about the server.
+func (bc *backendContract) Stats(ctx context.Context, params *ServerStatsParams) (*ServerStatsResult, error) {
```
Comment (rumyantseva) [inline internal/backends/backend.go:148]: Ok, I think I'm getting there :) So we have `Stats` for collections, databases, and server. `Stats` for collection and database are pretty similar (e.g. fields are similar with the same meaning), but stats for server is a totally different thing... Should it be called `Status` then? Or is it used in `msg_<something>stats` as well and this is why it's called `Stats`?

File: internal/handlers/sqlite/msg_datasize.go (hunk 1 above)
Comment (rumyantseva) [reply internal/handlers/sqlite/msg_datasize.go:89]: `Backend.ServerStats` is in fact `Backend.ServerStatus`, right? As it doesn't have the same fields as `CollectionStats` and `DatabaseStats`?

Comment (chilagrow) [reply internal/handlers/sqlite/msg_datasize.go:89]: Yes sure `Backend.ServerStatus` is more accurate :pray:

Comment (rumyantseva) [reply internal/handlers/sqlite/msg_datasize.go:89]: And I think for collections and databases I like the current naming - `Collection.Stats` and `Database.Stats` as we don't duplicate things in naming :) It's a bit confusing in the beginning, but I think it's fine.

File: internal/backends/backend.go (hunk 2 above)
Comment (chilagrow) [reply internal/backends/backend.go]: Let's use serverStatus, update the name.

File: internal/handlers/sqlite/msg_datasize.go (hunk 1 above)
Comment (chilagrow) [reply internal/handlers/sqlite/msg_datasize.go:89]: Yes true, lets leave those naming as it was :+1:

**Review by rumyantseva — APPROVED** [review]
LGTM!

**Review by AlekSi — APPROVED** [review]

**Pull request merged**